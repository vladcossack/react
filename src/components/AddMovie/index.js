import React from 'react';
import { AddMovie } from '../styled.js';

const AddMovieButton = ({showModal}) => (
  <AddMovie onClick={showModal}>+ ADD MOVIE</AddMovie>
);

export default AddMovieButton;