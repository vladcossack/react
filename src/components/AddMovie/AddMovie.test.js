import React from 'react';
import { render, screen } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';

import AddMovie from '../AddMovie';

describe('AddMovie component', () => {
  it('should render AddMovie snapshot', () => {
    const { asFragment } = render(<AddMovie />);
    expect(asFragment(<AddMovie />)).toMatchSnapshot();
  });

  it('should call showModal with positive result', () => {
    const showModal = jest.fn();
    render(<AddMovie showModal={showModal} />);

    const agreeBtn = screen.getByText('+ ADD MOVIE');

    fireEvent.click(agreeBtn);

    expect(showModal).toHaveBeenCalled();
  });
}); 