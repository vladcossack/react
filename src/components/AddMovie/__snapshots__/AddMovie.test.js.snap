// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie component should render AddMovie snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm kMBfSm"
  >
    + ADD MOVIE
  </button>
</DocumentFragment>
`;
